bloomfilter/src/Data/BloomFilter/Easy.hs

```diff
@@ -88,7 +88,7 @@ safeSuggestSizing
     -> Double           -- ^ desired false positive rate (0 < /e/ < 1)
     -> Either String (Word64, Int)
 safeSuggestSizing (fromIntegral -> capacity) errRate
-    | capacity <= 0                = Left "invalid capacity"
+    | capacity < 0                 = Left "invalid capacity"
```

Comment:
Note, the size will be always at least one though: https://github.com/IntersectMBO/lsm-tree/blob/fbd59d98477cbb684c581911096a60d38cca39e3/bloomfilter/src/Data/BloomFilter/Mutable.hs#L67

This is worth a comment why the negative capacity is allowed

Reply:
> Note, the size will be always at least one

In fact, with this function, it's at least 63. For very small `n`, creating a bloom filter is not super useful. I could add a comment like this:

-- Note that creating bloom filter with extremely small capacity is
-- often not very useful. This function will always suggest at least
-- 63 bits, even if the capacity is 0.

> negative capacity

Do you mean zero?

Reply:
> Do you mean zero?

Yes.

---

But I also wondere why not simply

```
| capacity <= 0 = Right (63, 1)
```

why allow zero, but not negative `Int`s?

Reply:
Except that `fromIntegral` converts argument to `Word64`, so there is nasty overflow happening, maybe

Reply:
Which `fromIntegral` do you mean? The ones here all convert to `Double` I think.

I'm leaning towards not allowing negative capacities. A bloom filter with 0 elements is not useful, but possible. A bloom filter with -1 elements is just not a thing.

Reply:
`take negative elements` is also not a thing, yet

```haskell
Prelude> take (-3) "foobar"
""
```

works.

Reply:
Makes sense, no strong preference either way. I'll change it.